src/pages/ResourceView.tsx

```diff
@@ -0,0 +1,29 @@
+import { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useLoaderData, useNavigate } from 'react-router-dom'
+import ResourceCard from 'src/components/Resources/ResourceCard'
+import { ResourceLoaderReturn } from 'src/loaders/resourceLoader'
+import Resource from 'src/types/Resource'
+
+export default function ResourceView() {
+  const navigate = useNavigate()
+  const { t } = useTranslation()
+  const data = useLoaderData() as ResourceLoaderReturn
+  const [resource, setResource] = useState(data.resource)
+  const onUpdateResource = (updatedResourceData: { resource: Resource }) => {
+    const updatedResource = updatedResourceData.resource
+    const newResource = { ...resource, ...updatedResource }
+
+    setResource(newResource)
```

Comment:
Might be nice to simplify to one line and save the variable declarations.
```suggestion
  setResource({ ...resource, ...updatedResourceData.resource })
```

Reply:
Updated in latest commit 👍